unit_testing/cli/test_cli_sct_register_to_template.py

```diff
@@ -62,3 +56,49 @@ def test_sct_register_to_template_non_rpi_data(tmp_path, template_lpi):
     img_reg = Image('template2anat.nii.gz')
     # Check if both images almost overlap. If they are right-left flipped, distance should be above threshold
     assert np.linalg.norm(img_orig.data - img_reg.data) < 1
+
+
+@pytest.mark.sct_testing
+@pytest.mark.usefixtures("run_in_sct_testing_data_dir")
+@pytest.mark.parametrize("fname_gt, remaining_args", [
+    ('template/template/PAM50_small_cord.nii.gz',
+     ['-l', 't2/labels.nii.gz', '-t', 'template', '-qc', 'qc-testing', '-param',
+      'step=1,type=seg,algo=centermassrot,metric=MeanSquares:step=2,type=seg,algo=bsplinesyn,iter=5,metric=MeanSquares']),
+    (os.path.join(__sct_dir__, 'data/PAM50/template/PAM50_cord.nii.gz'),
+     ['-ldisc', 't2/labels.nii.gz', '-ref', 'subject'])
+])
+def test_sct_register_to_template_dice_coefficient_against_groundtruth(fname_gt, remaining_args):
+    """Run the CLI script and verify transformed images have expected attributes."""
+    fname_seg = 't2/t2_seg-manual.nii.gz'
+    dice_threshold = 0.9
+    sct_register_to_template.main(argv=['-i', 't2/t2.nii.gz', '-s', fname_seg] + remaining_args)
+
+    # apply transformation to binary mask: template --> anat
+    sct_apply_transfo.main(argv=[
+        '-i', fname_gt,
+        '-d', fname_seg,
+        '-w', 'warp_template2anat.nii.gz',
+        '-o', 'test_template2anat.nii.gz',
+        '-x', 'nn',
+        '-v', '0'])
+
+    # apply transformation to binary mask: anat --> template
+    sct_apply_transfo.main(argv=[
+        '-i', fname_seg,
+        '-d', fname_gt,
+        '-w', 'warp_anat2template.nii.gz',
+        '-o', 'test_anat2template.nii.gz',
+        '-x', 'nn',
+        '-v', '0'])
+
+    # compute dice coefficient between template segmentation warped to anat and segmentation from anat
+    im_seg = Image(fname_seg)
+    im_template_seg_reg = Image('test_template2anat.nii.gz')
+    dice_template2anat = msct_image.compute_dice(im_seg, im_template_seg_reg, mode='3d', zboundaries=True)
```

Comment:
No reason to overcomplicate this

```diff
-     dice_template2anat = msct_image.compute_dice(im_seg, im_template_seg_reg, mode='3d', zboundaries=True)
+     dice_template2anat = spinalcordtoolbox.image.compute_dice(im_seg, im_template_seg_reg, mode='3d', zboundaries=True)
```

Reply:
At that point, it's not a CLI test at all! Might as well move it to the API tests, hehe.